w1r3/java/Dockerfile

```diff
@@ -0,0 +1,31 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM alpine:3.19 AS build
+
+# Install the minimal development tools:
+RUN apk update && \
+    apk add maven openjdk11
+
+WORKDIR /var/tmp/build
+COPY pom.xml /var/tmp/build/
+RUN mvn dependency:resolve-plugins
+RUN mvn dependency:resolve
+RUN mvn install
```

Comment:
This can be relaxed to `mvn package`. We don't need the built artifacts to be "installed" into the disk local maven repo because our project does not have cyclic dependencies like the google-cloud-java packages do.

Reply:
This is an optimization for GCB builds. Kaniko will cache this layer, so the next build will not need to download the dependencies again (until they change).

Reply:
The `dependency:resolve-plugins` and `dependency:resolve` will place the dependencies into the disk local maven repo, the `install` here is only to take the built package(s) for this `pom.xml` and copy them into the disk local maven repo. I don't think we need the empty jar and pom installed here, as we don't need it to be cached for the next build.